src/couch/src/couch_httpd_multipart.erl

```diff
@@ -104,6 +104,10 @@ mp_parse_atts(eof, {Ref, Chunks, Offset, Counters, Waiting}) ->
             receive
                 abort_parsing ->
                     ok;
+                {hello_from_writer, Ref, WriterPid} ->
+                    WriterRef = erlang:monitor(process, WriterPid),
+                    NewCounters = orddict:store(WriterPid, {WriterRef, 0}, Counters),
+                    mp_parse_atts(eof, {Ref, Chunks, Offset, NewCounters, Waiting});
```

Comment:
It's only 2 lines but noticed it repeats 3 times, wonder if it would look simpler with a tiny helper function like `{WRef, NewCounters} = writer_init(WriterPid, Counters)`. But then again, it's smaller enough so it doesn't look a lot better, maybe not worth bothering.

Reply:
I was on the fence but if we're adding extra logic to do at-most-once monitoring then a function probably does make sense (and could be refactored to share more in common with update_writer)

Reply:
I tried the idea and the pulled the logic into a separate function.